### PR TITLE
fix: adding passport:keys to post-autoload-dump

### LINKED ISSUE #4612: Laravel Passport's oauth-public.key is missing on Heroku
Some AJAX features fail to execute on Heroku environment with the following message:
```
Key path \"file:///app/storage/oauth-public.key\" does not exist or is not readable
```

I'm currently trying to solve the problem by calling `php artisan passport:keys` during the *release* (build) phase of the dyno lifecycle as mentionned in the procfile here:
```
release: php artisan monica:update --force -vvv
```

So basically, I'm trying to get the `monica:update` command to trigger `passport:keys` and then redeploy on Heroku....

Stay tuned :+1:

### PR DESCRIPTION
**Pull request by bdelespierre**
fixes #4612 

#### Before submitting the PR
- [x] Read the [CONTRIBUTING document](https://github.com/monicahq/monica/blob/master/CONTRIBUTING.md) before submitting your PR.
- [x] If the PR is related to an issue or fix one, don't forget to indicate it.
- [x] Create your PR as draft if it is not final yet. Mark it as ready... when it’s ready. Otherwise the PR will be considered complete and rejected if it's not working.

### General checks
- [x] Make sure that the change you propose is the smallest possible.
- [x] The name of the PR should follow the [conventional commits guideline](https://github.com/monicahq/monica/blob/master/docs/contribute/readme.md#conventional-commits) that the project follows.

#### Other tasks
- [ ] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG.md) entry added, if necessary, under `UNRELEASED`.
- [x] [CONTRIBUTORS](https://github.com/monicahq/monica/blob/master/CONTRIBUTORS) entry added, if necessary.
- [ ] If it's relevant and worth mentioning, create a changelog entry for this change. The changelog entry will appear inside the UI for all users to see. To know if your change is worth the creation of a changelog entry, [read the documentation](https://github.com/monicahq/monica/blob/master/docs/administrators/tips.md#when-is-it-relevant-to-create-a-changelog-entry).
- [ ] Don't forget to [ask for a free account](mailto:regis@monicahq.com) on https://monicahq.com as anyone who contributes can request a free account.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -115,7 +115,8 @@
         ],
         "post-autoload-dump": [
             "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
-            "@php artisan package:discover"
+            "@php artisan package:discover",
+            "@php artisan passport:keys"
```
Comment (asbiin) [inline composer.json:119]: I've made a special command to handle this.
This way if the files are already here, or if `PASSPORT_PRIVATE_KEY` and `PASSPORT_PUBLIC_KEY` env variable are present, the key will not be overwritten.

```suggestion
            "@php artisan monica:passport"
```

**Review by asbiin — CHANGES_REQUESTED** [review]
Thank you @bdelespierre that's a great contribution!
See my comment: I'm just concerned the key can be overwritten on each deploy, which is not a good behavior.

File: composer.json (hunk 1 above)
Comment (bdelespierre) [reply composer.json:119]: Okay I'll try this!

**Comment by bdelespierre** [comment]
```
remote:        > @php artisan monica:passport
remote:        **************************************
remote:        *     Application In Production!     *
remote:        **************************************
remote:        
remote:         Do you really wish to run this command? (yes/no) [no]:
remote:         > Command Canceled!
```

I think `-f` is missing

**Comment by bdelespierre** [comment]
@asbiin seems to work properly now
```
remote:        > @php artisan monica:passport --force
remote:        ✓ Creating encryption keys
```

**Comment by bdelespierre** [comment]
It breaks the CI tho...

image

**Comment by bdelespierre** [comment]
Long story short - we're better off registering the following environment variable in Heroku's dyno's settings

```
PASSPORT_PRIVATE_KEY
PASSPORT_PUBLIC_KEY
```

waaaaaay easier.

**Comment by crushgh** [comment]
I am still getting the ```Key path "file:///app/storage/oauth-private.key" does not exist or is not readable```  error - in the light of the above, what do I need to do? Do I just need to update Monica?

**Pull request closed without merging**